pkg/clientcert/certificate_test.go

```diff
@@ -161,10 +161,24 @@ func TestIsCertificateValid(t *testing.T) {
 
 func TestGetCertValidityPeriod(t *testing.T) {
 	certs := []byte{}
-	certs = append(certs, testinghelpers.NewTestCert("cluster0", 10*time.Second).Cert...)
+	firstCert := testinghelpers.NewTestCert("cluster0", 10*time.Second).Cert
+	certs = append(certs, firstCert...)
 	secondCert := testinghelpers.NewTestCert("cluster0", 5*time.Second).Cert
 	certs = append(certs, secondCert...)
-	expectedCerts, _ := certutil.ParseCertsPEM(secondCert)
+
+	firstCerts,_:= certutil.ParseCertsPEM(firstCert)
+	secondCerts,_ := certutil.ParseCertsPEM(secondCert)
+
+	fc := firstCerts[0]
+	sc := secondCerts[0]
+
+	var expectedCert *x509.Certificate
+	if fc.NotAfter.Before(sc.NotAfter) {
```

Comment:
this is confusion, could we change the sequence of `firstCert` and `secondCert` to make the `firstCert` always has a short duration, then use it as the `expectedCert`

Reply:
Yes, and I guess in this way we would have less code